core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -221,7 +221,9 @@ object Client {
     val conf = new SparkConf()
     val driverArgs = new ClientArguments(args)
 
-    conf.set("spark.rpc.askTimeout", "10")
+    if (!conf.contains("spark.rpc.askTimeout")) {
```

Comment:
OK, to move forward on this: @rxin I think this ends up affecting application launch because this is the bit that runs applications rather than starts workers, if that's what you mean. It will process `--conf` from the user command line but then override it here. The report in the JIRA seems to confirm that. At the least, this change would not affect the current behavior.

@andrewor14 that makes sense, though I note that the other `spark.rpc` configs are not in `config/package.scala`. The `ConfigBuilder` doesn't quite seem able to express a property whose default is another property either. Should I still try to refactor that?